pgjdbc/src/main/java/org/postgresql/jdbc/PgBlobBytea.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+/*
+ * Implementation of Blob interface backed by Postgres bytea fields.
+ *
+ * See PgBlob.java for implementation using Postgres Large Objects.
+ */
+
+package org.postgresql.jdbc;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+public class PgBlobBytea implements java.sql.Blob {
+
+  private byte[] data;
+
+  public static final int MAX_BYTES = 1073741824; // 1GiB max size of a bytea
```

Comment:
is there a reason for this to be public ?

Reply:
It's used in an error check (that I need to complete) in PgPreparedStatement

Reply:
if possible protected, even better private.